lib/wurk/scheduled.rb

```
@@ -0,0 +1,182 @@
+        @mutex = ::Mutex.new
+        @sleeper = ::ConditionVariable.new
+        @thread = nil
+        @rnd = ::Random.new
+        @last_cleanup_ms = 0
+      end
+
+      # Spawns the scheduler thread. INITIAL_WAIT delays the first sweep so
+      # a fleet-wide deploy doesn't have every freshly-booted process hit
+      # Redis simultaneously.
+      def start
+        @thread ||= safe_thread('scheduler') do # rubocop:disable Naming/MemoizedInstanceVariableName
```

RuboCop (1.86.2)

[warning] Unnecessary disabling of `Naming/MemoizedInstanceVariableName`.

(Lint/RedundantCopDisableDirective)